### PR TITLE
[JENKINS-60102] In order to capture the right window handle to get back to, confirmAlert/handleAlert should take a Runnable.

### PR DESCRIPTION
**Pull request by Vlatombe**
Also honors the provided timeout (#542 changed it to hardcoded 10 seconds)

### CODE REVIEW

File: src/main/java/org/jenkinsci/test/acceptance/po/CapybaraPortingLayerImpl.java [hunk 1]
```diff
@@ -372,13 +372,22 @@ public void check(String locator) {
...
+
+    public void runThenHandleAlert(Runnable runnable, Consumer<Alert> action) {
+        runThenHandleAlert(runnable, action, 10);
+    }
 
+    public void runThenHandleAlert(Runnable runnable, Consumer<Alert> action, int timeoutSeconds) {
+        String oldWindow = driver.getWindowHandle();
+        if (runnable != null) {
+            runnable.run();
+        }
```
Comment (Vlatombe) [inline src/main/java/org/jenkinsci/test/acceptance/po/CapybaraPortingLayerImpl.java]: This is the critical section

**Comment by olivergondza** [comment]
@Vlatombe, I see the `core.FormValidationTest.validate` is still failing. Is that expected?

https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/PR-547/1/

**Comment by Vlatombe** [comment]
I'll check tomorrow

Le lun. 11 nov. 2019 à 09:57, Oliver Gondža <notifications@github.com> a
écrit :

**Comment by Vlatombe** [comment]
Works on chrome but fails on firefox (the alert is not displayed)...

**Comment by Vlatombe** [comment]
Same failures on master - https://ci.jenkins.io/job/Core/job/acceptance-test-harness/job/master/198/testReport/

**Comment by Vlatombe** [comment]
Seems like some problem with event handling in firefox driver, when running the same manually the alert shows fine.

**Comment by olivergondza** [comment]
Cool, thanks! Btw, I am trying to see what would it take to switch to chrome for good: https://github.com/jenkinsci/acceptance-test-harness/pull/549

**Comment by olivergondza** [comment]
@Vlatombe, just for the record, the failure does not seem to be firefox specific as it is failing in chrome too. More like container specific...

**Comment by Vlatombe** [comment]
Ah, strange, I was running both locally

**Pull request merged**